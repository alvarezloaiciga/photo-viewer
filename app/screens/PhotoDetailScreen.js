import React, { Component } from 'react';
import { StyleSheet, Image } from 'react-native';

export default class PhotosScreen extends Component {
  static navigationOptions = {
    title: 'Photo Detail',
  };

  render() {
    const { params } = this.props.navigation.state;

    return(
      <Image source={{uri: params.photo.image_url}} style={styles.image} />
    )
  }
}


const styles = StyleSheet.create({
  image: {
    flex: 1,
    resizeMode: 'cover'
  },
})
