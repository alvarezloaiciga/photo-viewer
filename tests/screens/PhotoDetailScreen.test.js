import 'react-native';
import React from 'react';
import PhotoDetailScreen from '../../app/screens/PhotoDetailScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <PhotoDetailScreen navigation={ {state: { params: { photo: { image_url: "https://image.jpeg" }}} }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
