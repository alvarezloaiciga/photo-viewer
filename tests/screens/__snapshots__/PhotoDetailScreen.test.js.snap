// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": "https://image.jpeg",
    }
  }
  style={
    Object {
      "flex": 1,
      "resizeMode": "cover",
    }
  }
/>
`;
